src/stack.js

```diff
@@ -2,27 +2,27 @@
 
 export default class Stack {
   constructor() {
-    this.dataStore = [];
-    this.top = 0;
+    this.arr = [];
+    this.total = 0;
 }
   push(element) {
-    this.dataStore = element
-    this.top++
+    this.arr[this.total++] = element
   }
 
   pop() {
-    this.dataStore[this.top-1];
+    return this.arr[--this.total-1]
```

Comment:
I can't tell if pop() actually removes the item from the collection or not. It seems like it is reducing the length property but the item returned might still exist in the array. Might want to just check and make sure you have a test case to see if it not only returns the correct item, but also removes it from the collection.